src/data-set/display.ts

```diff
@@ -0,0 +1,233 @@
+    indentLevel + 1
+  )
+  log.indent(
+    `Max piece size: ${formatBytes(BigInt(dataSet.provider.products.PDP?.data?.maxPieceSizeInBytes ?? 0))}`,
+    indentLevel + 1
+  )
+  log.indent(
+    `Storage price: ${formatStoragePrice(dataSet.provider.products.PDP?.data?.storagePricePerTibPerDay ?? 0n)}`,
+    indentLevel + 1
+  )
+  log.indent(
+    `Min proving period: ${dataSet.provider.products.PDP?.data?.minProvingPeriodInEpochs ?? 0} epochs`,
+    indentLevel + 1
+  )
+  log.indent(`Location: ${dataSet.provider.products.PDP?.data?.location ?? 'unknown'}`, indentLevel + 1)
+  log.indent(
+    `Payment token: ${formatPaymentToken(dataSet.provider.products.PDP?.data?.paymentTokenAddress ?? 'unknown')}`,
+    indentLevel + 1
+  )
+  log.indent(`Commission: ${formatCommission(dataSet.commissionBps)}`, indentLevel + 1)
+  log.line('')
+}
+
+function renderPaymentDetails(dataSet: DataSetSummary, indentLevel: number = 0): void {
+  log.indent(pc.bold('Payment'), indentLevel)
+  log.indent(`PDP rail ID: ${dataSet.pdpRailId}`, indentLevel + 1)
+  if (dataSet.withCDN) {
+    log.indent(`CDN rail ID: ${dataSet.cdnRailId > 0 ? dataSet.cdnRailId : 'none'}`, indentLevel + 1)
+    log.indent(`Cache-miss rail ID: ${dataSet.cacheMissRailId > 0 ? dataSet.cacheMissRailId : 'none'}`, indentLevel + 1)
+  }
+  log.indent(`Payer: ${dataSet.payer}`, indentLevel + 1)
+  log.indent(`Payee: ${dataSet.payee}`, indentLevel + 1)
+  if (dataSet.pdpEndEpoch > 0) {
+    log.indent(pc.yellow(`PDP payments ended @ epoch ${dataSet.pdpEndEpoch}`), indentLevel + 1)
+  }
+  log.line('')
+}
+
+/**
+ * Render metadata key-value pairs with consistent indentation.
+ */
+function renderMetadata(metadata: Record<string, string>, indentLevel: number = 1, title: string = 'Metadata'): void {
+  log.indent(pc.bold(title), indentLevel)
+  const entries = Object.entries(metadata)
+  if (entries.length === 0) {
+    log.indent(pc.gray('none'), indentLevel + 1)
+    log.line('')
+    return
+  }
+
+  for (const [key, value] of entries) {
+    if (key === METADATA_KEYS.WITH_IPFS_INDEXING) {
+      log.indent(`${key}: ${pc.green('true')}`, indentLevel + 1)
+      continue
+    }
+    if (key === 'source' && value === 'filecoin-pin') {
+      log.indent(`${key}: ${pc.green(value)}`, indentLevel + 1)
+      continue
+    }
+    const displayValue = value === '' ? pc.gray('(empty)') : value
+    log.indent(`${key}: ${displayValue}`, indentLevel + 1)
+  }
+  log.line('')
+}
+
+/**
+ * Render a single piece entry including CommP, root CID, size, and extra metadata.
+ */
+function renderPiece(piece: PieceInfo, baseIndentLevel: number = 2): void {
+  const rootDisplay = piece.rootIpfsCid ?? piece.metadata?.[METADATA_KEYS.IPFS_ROOT_CID] ?? pc.gray('unknown')
+  const sizeDisplay = piece.size != null ? formatFileSize(piece.size) : pc.gray('unknown')
+
+  log.indent(pc.bold(`#${piece.pieceId}`), baseIndentLevel)
+  log.indent(`CommP: ${piece.pieceCid}`, baseIndentLevel + 1)
```

Comment:
```suggestion
  log.indent(`PieceCID: ${piece.pieceCid}`, baseIndentLevel + 1)
```

could do "PieceCID (CommP)" if you're trying to make it more different from "IPFS Root CID"

I always reach for "CommP" but we probably should more formally align on "PieceCID" as what we tell people